### ISSUE TITLE
Calling clearStorageData on a session with caching disabled kills the whole app without any error message/status

### ISSUE THREAD

**Issue by misternerd** [electron/electron#18585]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v5.0.2
* **Operating System:**
  * Ubuntu 16.04 x64
  * Ubuntu 18.04 x64

### Expected Behavior
The callback provided to `clearStorageData` should be called. Or an error should be thrown that clearing the storage isn't possible if `cache: false` is set on the session.

### Actual Behavior
The whole Electron app crashes, without any error message at all. The exit code is `0`.

### To Reproduce
This example crashes the process on my machine every time:
```javascript
const {app, BrowserWindow, session, Session} = require('electron');

app.on('ready', () => {
	const sess = session.fromPartition('some-new-partition', {
		cache: false,
	});
	sess.clearCache(() => logger.info('Cleared cache'));
	sess.clearStorageData({}, () => console.log('Storage cleared'));
	const window = new BrowserWindow({
		webPreferences: {
			session: sess
		}
	});
	window.loadURL('https://electronjs.org');
	window.show();
});
```
If `clearCache` is omitted, the app works as expected, i.e. the window is shown. But if `clearCache` isn't omitted, the app crashes at `clearStorageData` without emitting any error or faulty exit code. In the project this bug occurred in, a crash reporter was present, but it failed to create any crash report for this error (might have other reasons, though).

**Comment by MarshallOfSound** [comment]
@misternerd This issue is fixed in the latest `6.0.0` betas.  I'd suggest upgrading to those.

**Comment by electron-triage** [comment]
@misternerd We are going to close this issue as it fixed in later versions, see above comment. We will be happy to reopen this issue if you are still experiencing this on later versions of Electron.
